### PR TITLE
[Work in progress] Translate docs/codebase-overview

### PR DESCRIPTION
**Pull request by ahmedkrmn**
### Translated subsections:
- [x] Section Description

- [x] Contribute to React

- [x] External Dependencies

- [x] Top-Level Folders

- [x] Colocated Tests

- [x] Warnings and Invariants

- [x] Development and Production

- [x] Flow

- [x] Dynamic Injection

- [x] Multiple Packages

- [x] React Core

- [x] Renderers

- [x] Reconcilers

- [x] Stack Reconciler

- [x] Fiber Reconciler

- [x] Event System

- [x] What Next?

### CODE REVIEW

**Comment by iRayan7** [comment]
Hi @ahmedkrmn 👋any updates on this?

**Comment by ahmedkrmn** [comment]
Hi @iRayan7, sorry for the delay, I've been very busy lately. 
I will continue working on it starting from this weekend.

**Comment by iRayan7** [comment]
@ahmedkrmn Thanks 🙏

**Comment by iRayan7** [comment]
👋 Hi @ahmedkrmn! any updates on this?

**Comment by ahmedkrmn** [comment]
@iRayan7 All is done ! 🚀
Very sorry for the long delay.

**Comment by mohshbool** [comment]
Hey, I'm sorry but I can't review this at the time. Would anyone else do it? Thanks :)

File: content/docs/codebase-overview.md [hunk 1]
```diff
@@ -194,9 +194,9 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
 
 لا يتم حزم المطابقات بشكل منفصل لأنه لا يوجد حاليًا أي واجهة برمجة تطبيقات عامة. بدلاً من ذلك ، يتم استخدامها حصريًا بواسطة عارضين مثل React DOM و React Native.
 
-### Stack Reconciler {#stack-reconciler}
+### مطابق المكدس {#stack-reconciler}
 
-The "stack" reconciler is the implementation powering React 15 and earlier. We have since stopped using it, but it is documented in detail in the [next section](/docs/implementation-notes.html).
+المطابق "المكدس" (stack reconciler) ههو تطبيق يقوم بتشغيل React 15 والإصدارات الأقدم. لقد توقفنا عن استخدامه منذ ذلك الحين، لكن تم توثيقه بالتفصيل في [القسم التالي](/docs/implementation-notes.html).
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
مطابق "المكدس" (stack reconciler) هو تطبيق يقوم بتشغيل React 15 والإصدارات الأقدم. لقد توقفنا عن استخدامه منذ ذلك الحين، لكن تم توثيقه بالتفصيل في [القسم التالي](/docs/implementation-notes.html).
```

File: content/docs/codebase-overview.md [hunk 2]
```diff
@@ -168,23 +168,23 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
 
 الكود الخاص بنواة React يقع في [`packages/react`](https://github.com/facebook/react/tree/master/packages/react) بشجرة المصدر (source tree). و هي متوفرة علي npm كحزمة [`react`](https://www.npmjs.com/package/react). يسمى بناء المتصفح المستقل باسم `react.js`، ويصدر (exports) عمومي (global) يسمى `React`.
 
-### Renderers {#renderers}
+### العارضون {#renderers}
 
-React was originally created for the DOM but it was later adapted to also support native platforms with [React Native](https://facebook.github.io/react-native/). This introduced the concept of "renderers" to React internals.
+تم إنشاء React في الأصل من أجل DOM ولكن تم تكييفها لاحقًا أيضًا لدعم الأنظمة الأساسية الأصلية مع [React Native](https://facebook.github.io/react-native/). قدم هذا مفهوم "العارضون" لدواخل React.
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
تم إنشاء React في الأصل من أجل DOM ولكن تم تكييفها لاحقًا أيضًا لدعم الأنظمة الأساسية الأصلية مع [React Native](https://facebook.github.io/react-native/). قدم هذا مفهوم "التصيير" لدواخل React.
```

File: content/docs/codebase-overview.md [hunk 3]
```diff
@@ -168,23 +168,23 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
...
-* [React Native Renderer](https://github.com/facebook/react/tree/master/packages/react-native-renderer) renders React components to native views. It is used internally by React Native.
-* [React Test Renderer](https://github.com/facebook/react/tree/master/packages/react-test-renderer) renders React components to JSON trees. It is used by the [Snapshot Testing](https://facebook.github.io/jest/blog/2016/07/27/jest-14.html) feature of [Jest](https://facebook.github.io/jest) and is available as [react-test-renderer](https://www.npmjs.com/package/react-test-renderer) npm package.
+* يعرض [React DOM Renderer](https://github.com/facebook/react/tree/master/packages/react-dom)  مكونات React DOM. يقوم بتنفيذ [top-level `ReactDOM` APIs](/docs/react-dom.html) و هي متوفرة كحزمة [`react-dom`](https://www.npmjs.com/package/react-dom) علي npm. يمكن استخدامه أيضًا كحزمة متصفح مستقلة تسمى `react-dom.js` و التي تصدر `ReactDOM` عمومي.
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
* تصيير [React DOM Renderer](https://github.com/facebook/react/tree/master/packages/react-dom)  مكونات React DOM. يقوم بتنفيذ [top-level `ReactDOM` APIs](/docs/react-dom.html) و هي متوفرة كحزمة [`react-dom`](https://www.npmjs.com/package/react-dom) علي npm. يمكن استخدامه أيضًا كحزمة متصفح مستقلة تسمى `react-dom.js` و التي تصدر `ReactDOM` عمومي.
```

File: content/docs/codebase-overview.md [hunk 4]
```diff
@@ -168,23 +168,23 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
...
+* يعرض [React Native Renderer](https://github.com/facebook/react/tree/master/packages/react-native-renderer) مكونات React إلى وجهات النظر الأصلية. يتم استخدامه داخليًا بواسطة React Native.
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
* تصيير [React Native Renderer](https://github.com/facebook/react/tree/master/packages/react-native-renderer) مكونات React إلى وجهات النظر الأصلية. يتم استخدامه داخليًا بواسطة React Native.
```

File: content/docs/codebase-overview.md [hunk 5]
```diff
@@ -168,23 +168,23 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
...
+* يعرض [React Test Renderer](https://github.com/facebook/react/tree/master/packages/react-test-renderer) مكونات React الي اشجار JSON. يتم استخدامه بواسطة ميزة [Snapshot Testing](https://facebook.github.io/jest/blog/2016/07/27/jest-14.html) الخاصة ب[Jest](https://facebook.github.io/jest) و هو متاح كحزمة [react-test-renderer](https://www.npmjs.com/package/react-test-renderer) علي npm.
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
* تصيير [React Test Renderer](https://github.com/facebook/react/tree/master/packages/react-test-renderer) مكونات React الي اشجار JSON. يتم استخدامه بواسطة ميزة [Snapshot Testing](https://facebook.github.io/jest/blog/2016/07/27/jest-14.html) الخاصة ب[Jest](https://facebook.github.io/jest) و هو متاح كحزمة [react-test-renderer](https://www.npmjs.com/package/react-test-renderer) علي npm. 
```

File: content/docs/codebase-overview.md [hunk 6]
```diff
@@ -198,21 +198,21 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
...
-* Ability to split interruptible work in chunks.
-* Ability to prioritize, rebase and reuse work in progress.
-* Ability to yield back and forth between parents and children to support layout in React.
-* Ability to return multiple elements from `render()`.
-* Better support for error boundaries.
+* القدرة على تقسيم العمل المنقطع في أجزاء.
+* القدرة على تحديد الأولويات، إعادة صياغة وإعادة استخدام العمل قيد التقدم.
+* القدرة على الخضوع ذهابًا وإيابًا بين الآباء والأطفال لدعم التخطيط في React.
+* القدرة على إرجاع عناصر متعددة من `render()`.
+* دعم أفضل لحدود الخطأ.
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
* دعم أفضل لحدود الخطأ (error boundaries).
```

File: content/docs/codebase-overview.md [hunk 7]
```diff
@@ -198,21 +198,21 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
...
 
-You can read more about React Fiber Architecture [here](https://github.com/acdlite/react-fiber-architecture) and [here](https://blog.ag-grid.com/inside-fiber-an-in-depth-overview-of-the-new-reconciliation-algorithm-in-react). While it has shipped with React 16, the async features are not enabled by default yet.
+يمكنك قراءة المزيد عن React Fiber Architecture [هنا](https://github.com/acdlite/react-fiber-architecture) و [هنا](https://blog.ag-grid.com/inside-fiber-an-in-depth-overview-of-the-new-reconciliation-algorithm-in-react). بينما يتم شحنها مع React 16، لا يتم تمكين ميزات المزامنة (async) بشكل افتراضي بعد.
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
يمكنك قراءة المزيد عن "React Fiber Architecture" [هنا](https://github.com/acdlite/react-fiber-architecture) و [هنا](https://blog.ag-grid.com/inside-fiber-an-in-depth-overview-of-the-new-reconciliation-algorithm-in-react). بينما يتم شحنها مع React 16، لا يتم تمكين ميزات المزامنة (async) بشكل افتراضي بعد.
```

File: content/docs/codebase-overview.md [hunk 8]
```diff
@@ -216,7 +216,7 @@ React عبارة عن [monorepo](https://danluu.com/monorepo/). يحتوي مس
 
 ### نظام الأحداث {#event-system}
 
-يطبق React نظام أحداث اصطناعي غير مذعن للعارضين ويعمل مع كل من React DOM و React Native. شفرة المصدر الخاصة به موجودة في [`packages/events`](https://github.com/facebook/react/tree/master/packages/events).
+يطبق React نظام أحداث اصطناعي غير مذعن للعارضين ويعمل مع كل من React DOM و React Native. شفرة المصدر الخاصة به موجودة في [`packages/events`](https://github.com/facebook/react/tree/master/packages/react-events).
```
Comment (3imed-jaberi) [inline content/docs/codebase-overview.md]:
```suggestion
يطبق React نظام أحداث اصطناعي محايد لتصيير ويعمل مع كل من React DOM و React Native. 

شفرة المصدر الخاصة به موجودة في [`packages/events`](https://github.com/facebook/react/tree/master/packages/react-events).
```

**Comment by ahmedkrmn** [comment]
@3imed-jaberi Thanks for all the time and effort you took to review the entire page. I've committed all the requested changes. Please recheck to see if there are any further changes that should be made.

**Pull request merged**